libraries/ESP8266WiFi/src/include/UdpContext.h

```diff
@@ -522,6 +528,90 @@ class UdpContext
         reinterpret_cast<UdpContext*>(arg)->_recv(upcb, p, srcaddr, srcport);
     }
 
+#if LWIP_VERSION_MAJOR == 1
+    /*
+     * Code in this conditional block is copied/backported verbatim from
+     * LwIP 2.1.2 to provide pbuf_get_contiguous.
+     */
+
+    static const struct pbuf *
+    pbuf_skip_const(const struct pbuf *in, u16_t in_offset, u16_t *out_offset)
+    {
+      u16_t offset_left = in_offset;
+      const struct pbuf *q = in;
+
+      /* get the correct pbuf */
+      while ((q != NULL) && (q->len <= offset_left)) {
+        offset_left = (u16_t)(offset_left - q->len);
+        q = q->next;
+      }
+      if (out_offset != NULL) {
+        *out_offset = offset_left;
+      }
+      return q;
+    }
+
+    u16_t
+    pbuf_copy_partial(const struct pbuf *buf, void *dataptr, u16_t len, u16_t offset)
```

Comment:
Does this have to be in the class definition? Can it be moved out into the cpp?

Reply:
Again, same as above.